app.py

```diff
@@ -0,0 +1,118 @@
+import carball
+
+import os
+from flask import Flask, flash, request, redirect, url_for, request
+from werkzeug.utils import secure_filename
+from google.protobuf.json_format import MessageToJson
+import json
+
+UPLOAD_FOLDER = './uploads'
+ALLOWED_EXTENSIONS = set(['replay', 'replay'])
+
+app = Flask(__name__)
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+
+def shutdown_server():
+    func = request.environ.get('werkzeug.server.shutdown')
+    if func is None:
+        raise RuntimeError('Not running with the Werkzeug Server')
+    func()
+
+app = Flask(__name__)
+
+@app.route("/")
+def load():
+    return open("./html/home.html", "r").read()
+@app.route("/end")
+def end():
+    shutdown_server()
+
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+	
+@app.route('/upload', methods = ['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        template = """
```

Comment:
do not put html in a python file
Either use a templeting engine or put the html file client side and have the server just send data.